tests/services/run_devices/pacbio/housekeeper_service/test_pacbio_housekeeper_service.py

```diff
@@ -0,0 +1,116 @@
+
+
+def test_store_files_in_housekeeper(
+    pac_bio_housekeeper_service: PacBioHousekeeperService,
+    expected_pac_bio_run_data: PacBioRunData,
+    pac_bio_sample_internal_id: str,
+    smrt_cell_internal_id: str,
+):
+    # GIVEN a PacBioRunData object and a PacBioHousekeeperService
+
+    # WHEN storing files in Housekeeper
+    pac_bio_housekeeper_service.store_files_in_housekeeper(expected_pac_bio_run_data)
+
+    # THEN a SMRT cell bundle is created
+    assert pac_bio_housekeeper_service.hk_api.get_latest_bundle_version(smrt_cell_internal_id)
+
+    # THEN a sample bundle is created
+    assert pac_bio_housekeeper_service.hk_api.get_latest_bundle_version(pac_bio_sample_internal_id)
+
+    # THEN all expected files are listed under the smrt cell bundle
+    smrt_bundle_files: list[File] = (
+        pac_bio_housekeeper_service.hk_api.get_files_from_latest_version(
+            bundle_name=smrt_cell_internal_id
+        )
+    )
+    assert smrt_bundle_files
+
+    # THEN all expected files are listed under the sample bundle
+    sample_bundle_files: list[File] = pac_bio_housekeeper_service.hk_api.get_latest_bundle_version(
+        bundle_name=pac_bio_sample_internal_id
+    ).files
+    assert sample_bundle_files
+
+
+@pytest.mark.parametrize(
+    "file_fixture, file_data_fixture",
+    [
+        ("pac_bio_ccs_report_file", "ccs_report_pac_bio_file_data"),
+        ("pac_bio_hifi_read_file", "pac_bio_hifi_reads_file_data"),
+    ],
+    ids=["ccs_report", "hifi_reads"],
+)
+def test_create_bundle_info(
+    pac_bio_housekeeper_service: PacBioHousekeeperService,
+    pac_bio_metrics: PacBioMetrics,
+    file_fixture: str,
+    file_data_fixture: str,
+    request: pytest.FixtureRequest,
+):
+    """Test the correct file data object is created for a file."""
+    # GIVEN a PacBioMetrics and a Housekeeper PacBioHousekeeperService objects
+
+    # GIVEN a file path
+    file: Path = request.getfixturevalue(file_fixture)
+
+    # WHEN creating the file data object for the file
+    file_data: PacBioFileData = pac_bio_housekeeper_service._create_bundle_info(
+        file_path=file, parsed_metrics=pac_bio_metrics
+    )
+
+    # THEN the method should return the correct PacBioFileData object
+    expected_file_data: PacBioFileData = request.getfixturevalue(file_data_fixture)
+    assert file_data == expected_file_data
+
+
+def test_store_housekeeper_file_not_found(
+    pac_bio_housekeeper_service: PacBioHousekeeperService,
+    expected_pac_bio_run_data: PacBioRunData,
+):
+    # GIVEN a PacBioRunData object and a PacBioHousekeeperService
+
+    # WHEN storing files in Housekeeper
+
+    # THEN an error is raised
+    with mock.patch.object(
+        PacBioRunFileManager, "get_files_to_store", side_effect=PostProcessingRunFileManagerError
+    ):
+        with pytest.raises(PostProcessingStoreFileError):
+            pac_bio_housekeeper_service.store_files_in_housekeeper(expected_pac_bio_run_data)
```

Comment:
Can this be clarified a bit? Since we add a error as a side effect it is quite obvious that an error is raised. I also think the mock should be explained in another` #GIVEN` line
```suggestion
def test_store_housekeeper_file_not_found(
    pac_bio_housekeeper_service: PacBioHousekeeperService,
    expected_pac_bio_run_data: PacBioRunData,
):
    # GIVEN a PacBioRunData object and a PacBioHousekeeperService
   
    # GIVEN an error in fetching files to store
    with mock.patch.object(
        PacBioRunFileManager, "get_files_to_store", side_effect=PostProcessingRunFileManagerError
    ):
            # WHEN storing files in Housekeeper
        with pytest.raises(PostProcessingStoreFileError):
            pac_bio_housekeeper_service.store_files_in_housekeeper(expected_pac_bio_run_data)
            # THEN the correct error is raised
```

Reply:
It tests the error propagation from the error handlers. Each error is specific to a module. Whereas errors within a module can vary from FileNotFound to ValueError etc. 

When an error is raised in a module it raise the module-assigned error.